drivers/i2c/i2c_ll_stm32.c

```diff
@@ -61,24 +61,44 @@ static int i2c_stm32_transfer(struct device *dev, struct i2c_msg *msg,
 	LL_I2C_Enable(i2c);
 
 	current = msg;
+	/*
+	 * Set I2C_MSG_RESTART flag on first message in order to send start
+	 * condition
+	 */
+	current->flags |= I2C_MSG_RESTART;
 	while (num_msgs > 0) {
-		unsigned int flags = 0;
-
-		if (current->len > 255)
-			return -EINVAL;
+		u8_t *next_msg_flags = NULL;
 
-		/* do NOT issue the i2c stop condition at the end of transfer */
 		if (num_msgs > 1) {
 			next = current + 1;
+			next_msg_flags = &(next->flags);
+
+			/*
+			 * Stop or restart condition between messages
+			 * of different directions is required
+			 */
 			if (OPERATION(current) != OPERATION(next)) {
-				flags = I2C_MSG_RESTART;
+				if (!(next->flags & I2C_MSG_RESTART)) {
+					return -EINVAL;
```

Comment:
This seems broken for me, this means if the messages given to the API is broken, this will leave the bus in an inconsistent state

Reply:
Updated. Thanks for catching this.